main.cc

```diff
@@ -420,12 +420,33 @@ int main(int ac, char** av) {
     // If --version is requested, print it out and exit immediately to avoid
     // Seastar-specific warnings that may occur when running the app
     bpo::variables_map vm;
-    bpo::store(bpo::command_line_parser(ac, av).options(app.get_options_description()).allow_unregistered().run(), vm);
+    auto parsed_opts = bpo::command_line_parser(ac, av).options(app.get_options_description()).allow_unregistered().run();
+    bpo::store(parsed_opts, vm);
     if (vm["version"].as<bool>()) {
         fmt::print("{}\n", scylla_version());
         return 0;
     }
 
+    fmt::print("Scylla version {} starting ...\n", scylla_version());
+    if (ac) {
+        fmt::print("command used: \"{}", av[0]);
+        for (int i = 1; i < ac; ++i) {
+            fmt::print(" {}", av[i]);
+        }
+        fmt::print("\"\n");
+    }
+
+    fmt::print("Parsed command line options: [{}]\n",
+        boost::algorithm::join(parsed_opts.options | boost::adaptors::transformed([] (const auto& opt) {
+            if (opt.value.empty()) {
+                return opt.string_key;
+            }
+
+            return (opt.string_key.empty() ?  "(positional) " : fmt::format("{}: ", opt.string_key)) +
+                        boost::algorithm::join(opt.value, " ");
+        }), ", ")
+    );
+
```

Comment:
Please extract the whole thing into some function.

Reply:
Done.

Reply:
I looked at the diff again and it seems that you meant everything from the "starting..." message to the parsed options. I misunderstood previously. Fixed now.

Reply:
Actually I meant just the option formatting, but what you did is good too.